### PR TITLE
chore: add generic math max/min

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change adds some no-brainer generic code and updates tooling so that the master builds, lints, etc with it.

## Test Plan
- [ ] pass tests

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/resourcemanagers/fair_share.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/determined-ai/determined/master/internal/sproto"
 	"github.com/determined-ai/determined/master/pkg/actor"
 	"github.com/determined-ai/determined/master/pkg/check"
+	"github.com/determined-ai/determined/master/pkg/mmath"
```
Comment (stoksc) [inline master/internal/resourcemanagers/fair_share.go:12]: "mmath" == "more math"

Comment (dzhu) [reply master/internal/resourcemanagers/fair_share.go:12]: [bikeshedding] Maybe `mathx` or `xmath`? I've seen a few uses of the `x` for modified/extended versions of things (`github.com/huandu/xstrings`, `github.com/jmoiron/sqlx`, `github.com/stretchr/objx`).

File: master/internal/resourcemanagers/provisioner/scale_decider.go [hunk 2]
```diff
@@ -239,21 +240,7 @@ func (s *scaleDecider) findInstancesToTerminate() sproto.TerminateDecision {
...
-	return b
-}
-
-func min(a, b int) int {
-	if a < b {
-		return a
-	}
-	return b
+	desiredNum = mmath.Min(desiredNum, s.maxInstanceNum-len(s.instances))
+	desiredNum = mmath.Max(desiredNum, s.minInstanceNum-len(s.instances))
```
Comment (dzhu) [inline master/internal/resourcemanagers/provisioner/scale_decider.go:244]: Could use `clamp` here too if you add it.

**Review by dzhu — APPROVED** [review]
Hooray!

File: master/pkg/logger/log_buffer.go [hunk 3]
```diff
@@ -42,12 +29,12 @@ func computeSlice(startID int, endID int, limit int, totalEntries int, capacity
 	selectTail := startID == -1
 
 	// Limit values to appropriate bounds.
-	startID = max(startID, max(0, totalEntries-capacity))
-	endID = min(endID, totalEntries)
+	startID = mmath.Max(startID, mmath.Max(0, totalEntries-capacity))
```
[the remark this answers is not in the document]
Comment (stoksc) [reply master/pkg/logger/log_buffer.go:32]: yes to this others, but i dont think this one is quite a clamp. if is it, pls show me how you would write/expected behavior of `mathx.Clamp`.

[the remark this answers is not in the document]
Comment (stoksc) [reply master/pkg/logger/log_buffer.go:32]: will just replace with ` mmath.Max(0, startID, totalEntries-capacity))`

**Pull request merged**